File: 19.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* fast=head;
        ListNode* low=head;
        for(int i=0;i<n;i++){
            fast=fast->next;
        }
        if(!fast){
            return head->next;
        }
        while(fast->next!=nullptr){
            fast=fast->next;
            low=low->next;
        }
        low->next=low->next->next;
        return head;
    }
};

File: 19.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func removeNthFromEnd(head *ListNode, n int) *ListNode {
    slow,fast:=head,head
    for i:=0;i<n;i++{
        fast=fast.Next
    }
    if fast==nil{
        head=head.Next
    }else{
        for fast.Next!=nil{
            fast=fast.Next
            slow=slow.Next
        }
        slow.Next=slow.Next.Next
    }
    return head
}